### PR TITLE
Removal of Mono references in the code

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #4415

Changes proposed in this pull request:
 - Remove Mono conditionals in the code

The intention with this PR is to have a clean cut of Mono implementation, in case someone wants to add Mono on master later. (However, changes and refactoring makes this difficult.)
The refactoring has been kept to a minimal to simplify verification of this PR (for instance variable handling in forms have been adapted for Mono, but that is a very minor optimization.)

Note that there are still some references to EnvUtils to RunningOnWindows/Unix that are kept, that are not too annoying and could be useful in the future.
 
What did I do to test the code and ensure quality:
 - Code review
 - Random testing

Has been tested on (remove any that don't apply):
 - Windows 10

### CODE REVIEW

**Comment by gerhardol** [comment]
Note: Travis will fail as the Windows tests will run in mono, requirement for travis succesful build will have to be removed. (Also #4430).

File: GitUI/CommandsDialogs/FormCheckoutBranch.cs [hunk 1]
```diff
@@ -168,45 +168,26 @@ private void ApplyLayout()
...
-                Width = tableLayoutPanel1.PreferredSize.Width + 40;
-            }
+            Width = tableLayoutPanel1.PreferredSize.Width + 40;
 
             for (var i = 0; i < controls1.Length; i++)
             {
                 var margin = controls1[i].Margin;
-                var height = controls1[i].Height + margin.Top + margin.Bottom;
-                _controls.Add(controls1[i], height);
+                var cheight = controls1[i].Height + margin.Top + margin.Bottom;
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormCheckoutBranch.cs]: Is there any reason we couldn't reuse `height` var?

Comment (gerhardol) [reply GitUI/CommandsDialogs/FormCheckoutBranch.cs]: The declaration of 'height' prior in the function is no longer in a separate block.
I would prefer the blocks to be independent still, but in this case it makes sense to reuse 'height'

**Comment by RussKie** [comment]
This should now build successfully once rebased

**Comment by gerhardol** [comment]
Travis builds are still required

**Comment by RussKie** [comment]
Travis file is removed from the master....

**Comment by gerhardol** [comment]
Travis removed as part of #4430 no longer needed. The GE builds still require them though...

**Comment by RussKie** [comment]
I think we need to reinstate .travis.yml but have it reconfigured not to run
https://docs.travis-ci.com/user/conditional-builds-stages-jobs/

**Comment by gerhardol** [comment]
I did the smallest .travis.yml I could to get around the problem (I do not have privileges to change the configuration). Still requires that Travis service is up.

**Comment by RussKie** [comment]
We still need travis integration for 2.x branches and PRs targeting those

**Comment by RussKie** [comment]
It appears we missed inverted conditions like `if (!EnvUtils.RunningOnWindows())`
E.g.
https://github.com/gitextensions/gitextensions/blob/4d5ec8ef406ed7fc1e4bd423a3432610ab63c4b6/GitUI/CommandsDialogs/FormBrowse.cs#L236-L240

**Comment by gerhardol** [comment]
> It appears we missed inverted conditions like if (!EnvUtils.RunningOnWindows())

I deliberately left those in the code "could be useful in the future" as stated in the description.
(In most situations, they will not be useful but I did not find the handling annoying at least.)

There are references to Unix and MacOS too.
(Don't remember by heart, but maybe 20 code paths.)

**Comment by RussKie** [comment]
Thanks, no worries then.

**Pull request merged**